homeassistant/components/coolmaster/config_flow.py

```diff
@@ -25,7 +25,13 @@
 
 MODES_SCHEMA = {vol.Required(mode, default=True): bool for mode in AVAILABLE_MODES}
 
-DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str, **MODES_SCHEMA})
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        **MODES_SCHEMA,
+        vol.Required(CONF_SWING_SUPPORT, default=True): bool,
```

Comment:
Why is this configured and not just read from the device (which I think it can looking at the code)?

Reply:
Maybe I'm not reading it right but it appears to me that the bridge knows if there is swing or not and you could use that? [here](https://github.com/OnFreund/pycoolmasternet-async/blob/master/pycoolmasternet_async/coolmasternet.py#L90)

Reply:
What you're linking to is exactly the bool that's being passed from here :)
The coolmaster API documentation does not show any way to get swing support other than per unit, and a single instance can have dozens of units.

Reply:
Alright. I think I get what you mean now. I would anyway call each unit in that case on init for each entity to get all properties and functionality. Seems as this is local polling so not sure why we would need to save on api requests likewise you only have to do it once per unit to get the full function if I understand well then.

Reply:
I don't mind to remove this new config flag. It was done to reduce the amount of requests when they are not needed (with the assumption that it's costly). 
@OnFreund , are you ok with this proposal, as I think you recommended otherwise.
I'll implement whatever is agreed.

Reply:
Ok. Leave it as is then but I guess then you also want to make the default `False` or this discussion does not really make sense.

Reply:
I agree - I also believe the default should be `False`: https://github.com/home-assistant/core/pull/82809#pullrequestreview-1211747599

Reply:
Would it work to query all the device information during the config flow and then store the option in the config entry data without asking the user?